### PR TITLE
[SPARK-31985][SS] Remove incomplete/undocumented stateful aggregation in continuous mode

### PR DESCRIPTION
**Pull request by HeartSaVioR**
### What changes were proposed in this pull request?

This removes the undocumented and incomplete feature of "stateful aggregation" in continuous mode, which would reduce 1100+ lines of code.

### Why are the changes needed?

The work for the feature had been stopped for over an year, and no one asked/requested for the availability of such feature in community. Current state for the feature is that it only works with `coalesce(1)` which force the query to read and process, and write in "a" task, which doesn't make sense in production.

The remaining code increases the work on DSv2 changes as well - that's why I don't simply propose reverting relevant commits - the code path has been changed due to DSv2 evolution.

### Does this PR introduce _any_ user-facing change?

Technically no, because it's never documented and can't be used in production in current shape.

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
cc. @tdas @zsxwing @gaborgsomogyi @xuanyuanking @uncleGen

File: sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala [hunk 1]
```diff
@@ -286,9 +286,7 @@ class StreamingQueryManager private[sql] (sparkSession: SparkSession) extends Lo
 
     (sink, trigger) match {
       case (table: SupportsWrite, trigger: ContinuousTrigger) =>
-        if (operationCheckEnabled) {
-          UnsupportedOperationChecker.checkForContinuous(analyzedPlan, outputMode)
-        }
+        UnsupportedOperationChecker.checkForContinuous(analyzedPlan, outputMode)
```
Comment (jose-torres) [inline sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala:289]: was removing the operationCheckEnabled flag intentional?

**Review by jose-torres — COMMENTED** [review]
Mostly looks good, just a question about one change that might have been accidental.

Thanks for taking care of this.

File: sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala (hunk 1 above)
Comment (HeartSaVioR) [reply sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala:289]: That's not intentional. Nice finding. Thanks! I'll revert.

**Comment by HeartSaVioR** [comment]
@jose-torres Would you mind going through another round of review? Thanks in advance!

**Comment by jose-torres** [comment]
LGTM. I don't have the repo fully set up on my new computer, so I'll try to find time to set it up and merge tomorrow. (Or you can do it if you want to try out your new committer powers; congrats btw!)

**Comment by HeartSaVioR** [comment]
Thanks for the reviewing and kind words :) I'll deal with merging.

**Comment by HyukjinKwon** [comment]
@HeartSaVioR, no big deal but let's make sure to mention which branch this PR went through as a comment.

**Comment by HeartSaVioR** [comment]
Ah thanks for noticing. I missed it. It's merged only in master branch.

**Pull request closed without merging**